### PR TITLE
Updates necessary for ngrok-rust 0.14.0

### PR DESCRIPTION
**Pull request by bobzilladev**

### CODE REVIEW

**Comment by bobzilladev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #47** <a href="https://app.graphite.dev/github/pr/ngrok/ngrok-python/47" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ngrok/ngrok-python/47?utm_source=stack-comment).

File: Cargo.toml [hunk 1]
```diff
@@ -12,20 +12,22 @@ crate-type = ["cdylib"]
...
 bytes = "1.3.0"
-futures = "0.3.26"
 lazy_static = "1.4.0"
 log = "0.4"
 # pin mio until all dependencies are also on windows-sys 0.48
 # https://github.com/microsoft/windows-rs/issues/2410#issuecomment-1490802715
 mio = { version = "=0.8.6" }
-ngrok = { version = "0.13.1" }
+# ngrok = { version = "0.14.0" }
+ngrok = { path="../ngrok-rust/ngrok/" }
```
[the remark this answers is not in the document]
Comment (bobzilladev) [reply Cargo.toml:21]: aye, against the rust branch

File: src/connect.rs [hunk 2]
```diff
@@ -163,25 +161,20 @@ pub fn connect(
...
-                addr_str = addr_str.split_once("://").unwrap().1.to_string();
-            }
-            if addr_str.starts_with("https://") {
-                addr_str = addr_str.split_once("://").unwrap().1.to_string();
-                assume_port = 443;
-            }
-            if !addr_str.starts_with(PIPE_PREFIX) && !addr_str.contains(':') {
-                addr_str = format!("{addr_str}:{assume_port}");
+            if !addr_str.contains(':') {
+                addr_str = format!("tcp://{addr_str}:80");
```
[the remark this answers is not in the document]
Comment (bobzilladev) [reply src/connect.rs:177]: Yeah, it's documented as such to copy behavior of the community supported library, in the readme at:
https://github.com/ngrok/ngrok-python/blob/main/README.md#connection

File: src/http.rs [hunk 3]
```diff
@@ -104,21 +124,23 @@ impl NgrokHttpTunnelBuilder {
         let mut oauth = OauthOptions::new(provider);
         if let Some(allow_emails) = allow_emails {
             for v in allow_emails.iter() {
-                oauth = oauth.allow_email(v);
+                oauth.allow_email(v);
```
[the remark this answers is not in the document]
Comment (bobzilladev) [reply src/http.rs:127]: Yeah, a big part of this change is that the rust options are now mutable, so the pattern on all the configurations changed a bit.

File: test/test.py [hunk 4]
```diff
@@ -138,7 +138,7 @@ async def test_pipe_socket(self):
         tunnel = await session.http_endpoint().listen()
```
[the remark this answers is not in the document]
Comment (bobzilladev) [reply test/test.py:138]: True! updated.

File: src/tunnel.rs [hunk 5]
```diff
@@ -327,9 +338,9 @@ impl NgrokTunnel {
         let af_unix = socket.getattr(intern!(py, "AF_UNIX"));
         if let Ok(af_unix) = af_unix {
             if self.get_family(py)?.as_ref(py).eq(af_unix)? {
-                // pipe
+                // unix socket
                 let sockname_str: &PyString = sockname.downcast(py)?;
-                self.forward(py, format!("pipe:{sockname_str}"))?;
+                self.forward(py, format!("unix:{sockname_str}"))?;
```
[the remark this answers is not in the document]
Comment (bobzilladev) [reply src/tunnel.rs]: True! updated.

File: src/connect.rs [hunk 6]
```diff
@@ -163,25 +161,20 @@ pub fn connect(
 ) -> PyResult<Py<PyAny>> {
     let kwargs: &PyDict = options.unwrap_or(PyDict::new(py));
     // decode address string
-    let mut addr_str = "localhost:80".to_string();
+    let mut addr_str = "tcp://localhost:80".to_string();
```
[the remark this answers is not in the document]
Comment (bobzilladev) [reply src/connect.rs]: True! updated.

**Review by Wonderful-Space — APPROVED** [review]

**Comment by bobzilladev** [comment]
Turned out to need this `ngrok-rust` change to keep using the `rustls` dependency, and not try and build native `openssl-sys`, which fails on the more esoteric platforms. https://github.com/ngrok/ngrok-rust/pull/114

**Pull request merged**